server/core/include/process_manager.hpp

```diff
@@ -0,0 +1,185 @@
+        cron_builder& post(std::function<void(std::any&)> f){ post_function_ = f; return *this; }
+
+        /// \brief Set the execution function to be run.
+        ///
+        /// \param[in] The main function to be executed
+
+        cron_builder& to_execute(std::function<void(std::any&)> f) noexcept { to_execute_ = f; return *this; }
+
+        /// \brief Set the interval in seconds between runs of this task. A value of Zero(the default), results
+        /// in the task being run every time the cron manager loops.
+        ///
+        /// \param[in] n The number of seconds between runs of the task
+        cron_builder& interval(int n) noexcept { interval_ = std::chrono::seconds(n); return *this; }
+
+        /// \brief Set the user defined data to be made available to the task in question
+        ///
+        /// \param[in] a The user defined data to be made available.
+        cron_builder& data(std::any& a) noexcept { data_ = a; return *this; }
+
+        /// \brief Sets whether or not post function should run more than once. Defaults to true.
+        ///
+        /// \param[in] a a bool indicating whether the post function should only run once.
+        cron_builder& run_post_once(bool a) noexcept { run_post_once_ = a; return *this; }
+
+        /// \brief Sets whether or not the pre function should be run every time the
+        /// main function is run. Defaults to true.
+        ///
+        /// \param[in] a a bool indicating whether or not the pre function should only run once.
+        cron_builder& run_pre_once(bool a) noexcept { run_pre_once_ = a; return *this; }
+
+        /// \brief Convert the cron_builder into a cron_task instance.
+        ///
+        /// \retval A cron_task object, ready to be inserted into the cron system.
+        cron_task finalize() const noexcept{
+            return cron_task(to_execute_,
+                             pre_function_,
+                             post_function_,
+                             interval_,
+                             run_pre_once_,
+                             run_post_once_,
+                             data_);
+        }
+    }; // struct cron_builder
+
+    /// \brief Management struct for the facility. A singleton.
+    struct cron{
+        /// \brief Loop across all the tasks in the cron structure
+        void run(){
+            for(auto& task: tasks) {
+                task.run();
+            }
+
+        }
+        /// \brief Get the single instance of the cron facility.
+        static cron* get() noexcept {
+            static cron singleton;
+            return &singleton;
+        }
+        /// \brief Add a task to the cron facility.
+        void add_task(cron_task&& ct){
+            tasks.emplace_back(std::move(ct));
+        }
+    private:
+        std::vector<cron_task> tasks;
+        cron(){}
+        cron(cron&) = delete;
+        cron(cron&&) = delete;
```

Comment:
missing:
- `cron& operator=(const cron&) = delete;`
- `cron& operator=(cron&&) = delete;`

you can disable all copy and move support by marking the copy constructor and copy assignment operator as deleted. for example:
```c++
cron(const cron&) = delete;
cron& operator=(const cron&) = delete;
```